Habitica/src/main/java/com/habitrpg/android/habitica/data/implementation/TaskRepositoryImpl.kt

```diff
@@ -146,6 +146,8 @@ class TaskRepositoryImpl(
                     bgTask.completed = up
                     if (TaskType.DAILY == bgTask.type && up) {
                         bgTask.streak = (bgTask.streak ?: 0) + 1
+                    } else if(TaskType.DAILY == bgTask.type && !up && (bgTask.streak != null && bgTask.streak!! >= 1)){
+                        bgTask.streak = bgTask.streak!! - 1

```

Comment:
Thanks for submitting - this fix works!
Would you be able to edit the statement to match the existing line that adds +1 to the streak? I haven't tested this but something like -
```
if (TaskType.DAILY == bgTask.type) {
                        if (up) {
                            bgTask.streak = (bgTask.streak ?: 0) + 1
                        } else {
                            bgTask.streak = (bgTask.streak ?: 0) - 1
                        }
                    }
```
This helps with readability and makes it a little more type-safey!

Reply:
Thank you for the advice, and fixed it!